darwin/utils.py

```diff
@@ -312,129 +290,131 @@ def parse_darwin_json(path: Path, count: Optional[int]) -> Optional[dt.Annotatio
         data = json.load(f)
         if "annotations" not in data:
             return None
-        if "fps" in data["image"] or "frame_count" in data["image"]:
-            return _parse_darwin_video(path, data, count)
+
+        if _parse_version(data).major == 2:
+            return _parse_darwin_v2(path, data)
         else:
-            return _parse_darwin_image(path, data, count)
+            if "fps" in data["image"] or "frame_count" in data["image"]:
+                return _parse_darwin_video(path, data, count)
+            else:
+                return _parse_darwin_image(path, data, count)
 
 
-@deprecation.deprecated(
-    deprecated_in="0.7.5",
-    removed_in="0.8.0",
-    current_version=__version__,
-    details="Use 'parse_darwin_json' instead",
-)
-def parse_darwin_image(path: Path, data: Dict[str, Any], count: Optional[int]) -> dt.AnnotationFile:
-    """
-    Parses the given JSON file in v7's darwin proprietary format. Works only for images.
+def _parse_darwin_v2(path: Path, data: Dict[str, Any]) -> dt.AnnotationFile:
+    item = data["item"]
+    item_source = item.get("source_info", {})
+    slots: List[dt.Slot] = list(filter(None, map(_parse_darwin_slot, item.get("slots", []))))
+    annotations: List[Union[dt.Annotation, dt.VideoAnnotation]] = _data_to_annotations(data)
+    annotation_classes: Set[dt.AnnotationClass] = set([annotation.annotation_class for annotation in annotations])
 
-    Parameters
-    ----------
-    path : Path
-        Path to the file to parse.
-    data : Dict[str, Any]
-        The decoded JSON file in Python format.
-    count : Optional[int]
-        Optional count parameter. Used only if the 's image sequence is None.
+    if len(slots) == 0:
+        annotation_file = dt.AnnotationFile(
+            version=_parse_version(data),
+            path=path,
+            filename=item["name"],
+            annotation_classes=annotation_classes,
+            annotations=annotations,
+            is_video=False,
+            image_width=None,
+            image_height=None,
+            image_url=None,
+            image_thumbnail_url=None,
+            workview_url=item_source.get("workview_url", None),
+            seq=0,
+            frame_urls=None,
+            remote_path=item["path"],
+        )
+    else:
+        slot = slots[0]
```

Comment:
this seems like a strange hard-coding. Are we ignoring other slots? how do we parse the annotations of slots other than the first?

Reply:
This has nothing to do with annotations here, this is just about metadata like file names etc. This is mostly backwards compatibility with V1 format, where we still take values from root of `AnnotationFile`. All V2 code is looking at `slots` directly and pull data from there.